### PR TITLE
Prototype Scoped Bean With Runtime Arguments

### PR DESCRIPTION
**Pull request by sachin071287**
https://drafts.baeldung.com/wp-admin/post.php?post=189380&action=edit

### CODE REVIEW

File: spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(loader = AnnotationConfigContextLoader.class, classes = EmployeeConfig.class)
+public class DynamicBeanUnitTest {
+
+    @Test
+    public void givenPrototypeBean_WhenFunction_ThenNewInstanceReturn() {
+
+        AbstractApplicationContext context = new AnnotationConfigApplicationContext(
+                EmployeeConfig.class);
```
Comment (KevinGilmore) [inline spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java:21]: Let's use four-space indentation (from previous line) for the continuation lines like this one and a few others noted below.

File: spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        Employee secondInstance = secondContext.getEmployee("kumar");
+
+        Assert.assertTrue(firstInstance != secondInstance);
+    }
+
+    @Test
+    public void givenPrototypeBean_WhenLookup_ThenNewInstanceReturn() {
+
+        AbstractApplicationContext context = new AnnotationConfigApplicationContext(
+                EmployeeConfig.class);
```
Comment (KevinGilmore) [inline spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java:34]: indentation

File: spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        EmployeeBeanUsingLookUp secondContext = context.getBean(EmployeeBeanUsingLookUp.class);
+        Employee firstInstance = firstContext.getEmployee("sachin");
+        Employee secondInstance = secondContext.getEmployee("kumar");
+        Assert.assertTrue(firstInstance != secondInstance);
+    }
+
+    @Test
+    public void givenPrototypeBean_WhenObjectProvider_ThenNewInstanceReturn() {
+        AbstractApplicationContext context = new AnnotationConfigApplicationContext(
+                EmployeeConfig.class);
```
Comment (KevinGilmore) [inline spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java:45]: indentation

File: spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        EmployeeBeanUsingObjectProvider firstContext = context
+                .getBean(EmployeeBeanUsingObjectProvider.class);
```
Comment (KevinGilmore) [inline spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java:47]: indentation

File: spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        EmployeeBeanUsingObjectProvider secondContext = context
+                .getBean(EmployeeBeanUsingObjectProvider.class);
```
Comment (KevinGilmore) [inline spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java:49]: indentation

**Review by KevinGilmore — COMMENTED** [review]
See notes about indentation.

File: spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java (hunk 1 above)
Comment (sachin071287) [reply spring-di-4/src/test/java/com/baeldung/prototypebean/dynamicarguments/DynamicBeanUnitTest.java]: updated

**Pull request merged**